### DISCUSSION TITLE
BirdNet Pi responds to web but doesn't record...

### DISCUSSION THREAD

**Discussion by bcardi0427** [mcguirepr89/BirdNET-Pi#703] [Q&A] [answered]
My BirdNet Pi was doing fine until yesterday evening and hasn't picked up anything since. I've rebooted several times and today it doesn't even show the logs. I have no idea what I should do? Maybe just reinstall it?

**Comment by Scraggarax** [comment]
This seems like something similar #653 as well as #698. And it's been reported a few times lately.

What I think might be happening, is that when the services are stopped for an update or a sudden loss of power, the analysis service just hangs or needs a restart.

I'm not entirely sure on this part, but sometimes it locks up if I'm not viewing the log as it does it at the same time.


Go into your services, and restart the analysis services. Then go to the log and watch the date and time on the right. You'll see where it's at and just keep an eye on that log until it caches up. If it's been offline for a few days, you're going to have to restart the analysis service a few times to get this working again.


Be sure to do this before your birdnet fills up with recordings. Once it does that, it will not have enough free space to do anything else. You're going to have to go into the file manager and delete the audio files from the birdsong folder to make enough space to do anything.

**Reply by ehpersonal38** [reply]
Thanks for your help! Yeah this is a known issue, a fix is being worked on but not sure the ETA on when it'll be finished, it requires a pretty big overhaul of the current structure of BirdNET-Pi.

**Comment by bcardi0427** [comment]
I've restarted the analysis service several times and all here is what
comes up, the last three lines are all I see in the log from web app.

birdnet_server.service - BirdNET Analysis Server
     Loaded: loaded
(/home/bcardi/BirdNET-Pi/templates/birdnet_server.service; enabled; vendor
preset: enabled)
     Active: active (running) since Thu 2023-01-19 15:48:28 EST; 1s ago
   Main PID: 13260 (python3)
      Tasks: 1 (limit: 1993)
     Memory: 7.0M
        CPU: 956ms
     CGroup: /system.slice/birdnet_server.service
             └─13260 /home/bcardi/BirdNET-Pi/birdnet/bin/python3
/usr/local/bin/server.py

Jan 19 15:48:28 birdnetpi systemd[1]: Started BirdNET Analysis Server.


● birdnet_server.service - BirdNET Analysis Server
     Loaded: loaded
(/home/bcardi/BirdNET-Pi/templates/birdnet_server.service; enabled; vendor
preset: enabled)
     Active: active (running) since Thu 2023-01-19 15:52:52 EST; 1s ago
   Main PID: 17189 (python3)
      Tasks: 1 (limit: 1993)
     Memory: 3.0M
        CPU: 792ms
     CGroup: /system.slice/birdnet_server.service
             └─17189 /home/bcardi/BirdNET-Pi/birdnet/bin/python3
/usr/local/bin/server.py

Jan 19 15:52:52 birdnetpi systemd[1]: Started BirdNET Analysis Server.
● birdnet_server.service - BirdNET Analysis Server
     Loaded: loaded
(/home/bcardi/BirdNET-Pi/templates/birdnet_server.service; enabled; vendor
preset: enabled)
     Active: active (running) since Thu 2023-01-19 16:00:25 EST; 1s ago
   Main PID: 24044 (python3)
      Tasks: 1 (limit: 1993)
     Memory: 2.8M
        CPU: 671ms
     CGroup: /system.slice/birdnet_server.service
             └─24044 /home/bcardi/BirdNET-Pi/birdnet/bin/python3
/usr/local/bin/server.py

Jan 19 16:00:25 birdnetpi systemd[1]: Started BirdNET Analysis Server.
● birdnet_server.service - BirdNET Analysis Server
     Loaded: loaded
(/home/bcardi/BirdNET-Pi/templates/birdnet_server.service; enabled; vendor
preset: enabled)
     Active: active (running) since Thu 2023-01-19 16:04:54 EST; 2s ago
   Main PID: 28078 (python3)
      Tasks: 1 (limit: 1993)
     Memory: 5.7M
        CPU: 1.405s
     CGroup: /system.slice/birdnet_server.service
             └─28078 /home/bcardi/BirdNET-Pi/birdnet/bin/python3
/usr/local/bin/server.py

Jan 19 16:04:54 birdnetpi systemd[1]: Started BirdNET Analysis Server.



16:06:54---INFO: Created TensorFlow Lite XNNPACK delegate for CPU.
16:06:54---INFO: Created TensorFlow Lite delegate for select TF ops.
16:06:54---INFO: TfLiteFlexDelegate delegate: 1 nodes delegated out of 182
nodes with 1 partitions.

**Reply by ehpersonal38** [reply]
Looks like your server is running but the client is not. I would restart all core services and see if that resolves your issue

**Comment by bcardi0427** [comment]
I think I've found out what my problem is and I've seen others that have had this same problem. In the advanced settings there is a setting for your sound device. I've tried several devices but my arecord -L results are below and I get this log error:

Jan 19 18:29:15 birdnetpi birdnet_recording.sh[192514]: + arecord -f S16_LE -c2 -r48000 -t wav --max-file-time 15 -D ''\''default'\''' --use-strftime /home/bcardi/BirdSongs/%B-%Y/%d-%A/%F-birdnet-%H:%M:%S.wav
Jan 19 18:29:15 birdnetpi birdnet_recording.sh[192523]: ALSA lib pcm.c:2660:(snd_pcm_open_noupdate) Unknown PCM 'default'
Jan 19 18:29:15 birdnetpi birdnet_recording.sh[192523]: arecord: main:830: audio open error: No such file or directory
Jan 19 18:29:15 birdnetpi systemd[1]: birdnet_recording.service: Main process exited, code=exited, status=1/FAILURE
Jan 19 18:29:15 birdnetpi systemd[1]: birdnet_recording.service: Failed with result 'exit-code'.



 Discard all samples (playback) or generate zero samples (capture)
lavrate
    Rate Converter Plugin Using Libav/FFmpeg Library
samplerate
    Rate Converter Plugin Using Samplerate Library
speexrate
    Rate Converter Plugin Using Speex Resampler
jack
    JACK Audio Connection Kit
oss
    Open Sound System
pulse
    PulseAudio Sound Server
upmix
    Plugin for channel upmix (4,6,8)
vdownmix
    Plugin for channel downmix (stereo) with a simple spacialization
default
usbstream:CARD=LIBRETECHCC
    LIBRETECH-CC
    USB Stream Output
hw:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Direct hardware device without any conversions
plughw:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Hardware device with all software conversions
sysdefault:CARD=S3
    Sound Blaster Play! 3, USB Audio
    Default Audio Device
front:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Front output / input
dsnoop:CARD=S3,DEV=0
    Sound Blaster Play! 3, USB Audio
    Direct sample snooping device
usbstream:CARD=S3
    Sound Blaster Play! 3
    USB Stream Output

I've tried default in that dialog and Sound Blaster Play! 3, USB Audio in there but it doesn't seem to want to work.

**Reply by ehpersonal38** [reply]
Try using this exactly as written:
`plughw:CARD=S3,DEV=0`

**Reply by bcardi0427** [reply]
Tried that and it doesn't look like it worked. On the System Info page the arecord near the bottom says Off.

**Reply by bcardi0427** [reply]
I'm thinking of just reinstalling again and see what happens.

**Reply by bcardi0427** [reply]
I finally got it working today. I had to use 'default' as the device as each time I restarted recording service it failed with everything else, I also had to do a core reset after changing to default and it looked like it took an hour before it started writing the results. I have no idea why, I'm going to look through the code and see if I can figure out why when I get some extra time. The big thing is, It now works.

**Comment by ehpersonal38** [comment, marked as answer]
Glad to hear that! If it fails again I would immediately scroll through the file `/var/log/syslog` until you find the moment in time where your analysis stopped, and look for any error messages that may point to the cause.
